contracts/APPCoin.sol

```diff
@@ -13,13 +13,34 @@ import "@openzeppelin/contracts/utils/introspection/IERC1820Registry.sol";
 contract APPCoin is Initializable, ERC777Upgradeable, PausableUpgradeable, OwnableUpgradeable, UUPSUpgradeable, IERC777Recipient {
     bytes32 private constant _TOKENS_RECIPIENT_INTERFACE_HASH = keccak256("ERC777TokensRecipient");
     address public apiCoin;
+    struct WeightEntry {
+        string name;
```

Comment:
named with `resourceId` for better clarity?

Reply:
It could be a string like '/v1/tokens', while id is more likely to be a number.